src/format/KdbxXmlReader.cpp

```diff
@@ -875,11 +875,13 @@ QPair<QString, QString> KdbxXmlReader::parseEntryBinary(Entry* entry)
     }
 
     if (keySet && valueSet) {
-        if (entry->attachments()->hasKey(key)) {
-            raiseError(tr("Duplicate attachment found"));
-        } else {
-            entry->attachments()->set(key, value);
+        if (entry->attachments()->hasKey(key) && entry->attachments()->value(key) != value) {
+            // NOTE: This only impacts KDBX 3.x databases
+            // Prepend a random string to the key to make it unique and prevent data loss
+            key = key.prepend(QUuid::createUuid().toString().mid(1, 8) + QStringLiteral("_"));
```

Comment:
I'd append it and I wouldn't use QStringLiteral just for a `"_"` literal. This code also doesn't necessarily guarantee unique keys. That would need another check if the generated key already exists.

Reply:
It's a rare, onetime occurrence if ever. Appending causes the attachment to be unopenable because it overrides the extension

Reply:
It's rare, but my OCD keeps kicking in. I don't want to risk data loss to avoid data loss, even if the chances are reduced.

Reply:
8 chars from a uuid is very unique when used almost never in the same entry. This isn't a global problem, it's per entry.

Reply:
Plus this only impacts kdbx 3 which I think we should make as deprecated.

Reply:
If its just per entry and not global, I can perhaps restrain and gag myself. But my feet are still itching.